src/main/scala-2.11/markets/MarketLike.scala

```diff
@@ -1,27 +1,49 @@
 package markets
 
 import akka.actor.ActorRef
+import akka.agent.Agent
 
+import markets.clearing.Fill
 import markets.clearing.engines.MatchingEngine
 import markets.orders.Order
+import markets.tickers.Tick
 import markets.tradables.Tradable
 
 /** Mixin Trait providing MarketLike behavior to some BaseActor. */
 trait MarketLike {
   this: BaseActor =>
 
-  def clearingMechanism: ActorRef
+  def matchingEngine: MatchingEngine
+
+  def settlementMechanism: ActorRef
+
+  def ticker: Agent[Tick]
 
   def tradable: Tradable
 
   def marketActorBehavior: Receive = {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Default case not specified.](https://www.codacy.com/app/davidrpugh/markets-sandbox/file/1595769168/issues/source?bid=2921361&fileBranchId=3026497#l24)